### PR TITLE
Improvements to align CTS and Spec for Loader

### PR DESCRIPTION
**Pull request by martygrant**
- Add a null pointer check for urLoaderConfigCreate
- Add a null pointer and invalid size check for urLoaderConfigGetinfo
- Add invalid size test for urLoaderConfigGetInfo
- Add reference count test for urLoaderConfigGetInfo
- Add new test for urLoaderConfigSetCodeLocationCallback
- Add new test for UR_LOADER_CONFIG_INFO_AVAILABLE_LAYERS to test the list is valid (empty or valid values)
- Updated doc wording to say no layers are enabled by default

### CODE REVIEW

File: test/loader/loader_config/urLoaderConfigGetInfo.cpp [hunk 1]
```diff
@@ -23,30 +23,115 @@ struct urLoaderConfigGetInfoTest
...
+    ASSERT_EQ(UR_RESULT_ERROR_INVALID_SIZE,
+              urLoaderConfigGetInfo(loaderConfig, infoType, 0,
+                                    infoAllocation.data(), &infoSize));
+
+    ASSERT_EQ(UR_RESULT_ERROR_INVALID_SIZE,
+              urLoaderConfigGetInfo(loaderConfig, infoType, 1,
+                                    infoAllocation.data(), &infoSize));
+}
+
+struct urLoaderConfigGetInfoTest : urLoaderConfigGetInfoWithParamTest {};
```
Comment (aarongreig) [inline test/loader/loader_config/urLoaderConfigGetInfo.cpp:68]: instead of this you can do `using urLoaderConfigGetInfoTest = LoaderConfigTest;` and then you don't need to do the `INSTANTIATE_TEST_SUITE` stuff, you can just define the tests like `TEST_F(urLoaderConfigGetInfoTest, nameOfTest) {`

File: test/loader/loader_config/urLoaderConfigGetInfo.cpp [hunk 2]
```diff
@@ -23,30 +23,115 @@ struct urLoaderConfigGetInfoTest
...
                                     nullptr, &infoSize));
 }
+
+TEST_P(urLoaderConfigGetInfoWithParamTest, InvalidSize) {
+    ASSERT_EQ(UR_RESULT_ERROR_INVALID_SIZE,
+              urLoaderConfigGetInfo(loaderConfig, infoType, 0,
+                                    infoAllocation.data(), &infoSize));
+
+    ASSERT_EQ(UR_RESULT_ERROR_INVALID_SIZE,
+              urLoaderConfigGetInfo(loaderConfig, infoType, 1,
```
Comment (aarongreig) [inline test/loader/loader_config/urLoaderConfigGetInfo.cpp:64]: rather than `1` for these cases it's better to do `infoSize - 1` or something similar, to guarantee that the size is incorrect

File: test/loader/loader_config/urLoaderConfigGetInfo.cpp (hunk 1 above)
Comment (martygrant) [reply test/loader/loader_config/urLoaderConfigGetInfo.cpp]: yeah googletest can be a bit awkward sometimes, think I done your suggestion in a later PR as well when I noticed this option. Updated it, cheers!

File: test/loader/loader_config/urLoaderConfigGetInfo.cpp (hunk 2 above)
Comment (martygrant) [reply test/loader/loader_config/urLoaderConfigGetInfo.cpp]: makes sense, updated to this, thanks!

**Review by aarongreig — APPROVED** [review]
